pylint/lint/parallel.py

```diff
@@ -165,6 +165,7 @@ def check_parallel(
             msg_status,
             mapreduce_data,
         ) in executor.map(_worker_check_single_file, files):
+            assert isinstance(base_name, str)
```

Comment:
This should now always be `str` without an assertion

Reply:
mypy does not like it:
```
pylint/lint/parallel.py:168: error: Incompatible types in assignment (expression has type "Optional[str]", variable has type "str")  [assignment]

Found 1 error in 1 file (checked 3 source files)
```

Reply:
The typing on `PyLinter` should be updated!

Reply:
I rebased so we can rebase and merge, there was quite a lot more to change than I thought :D